core/arch/arm/mm/core_mmu.c

```diff
@@ -1927,6 +1927,27 @@ TEE_Result core_mmu_remove_mapping(enum teecore_memtypes type, void *addr,
 	return TEE_SUCCESS;
 }
 
+struct tee_mmap_region *core_mmu_find_mapping_exclusive(enum
+							teecore_memtypes type,
+							size_t len)
+{
+	unsigned int map_cnt = 0;
```

Comment:
This function could be simplified as (skipping `map_cnt`, exit early and avoiding the ternary operator at the end):
``` C
        struct tee_mmap_region *map = NULL; 
        struct tee_mmap_region *map_found = NULL;
        
        if (!len)
                return NULL;
        
        for (map = get_memory_map(); !core_mmap_is_end_of_table(map); map++) {
                if (map->type != type)
                        continue;
                
                if (map_found) 
                        return NULL;
                map_found = map;
        }

        if (!map_found || map_found->size < len)
                return NULL;

        return map_found;
```